### PR TITLE
MIDRC-847 GitHub workflows + add "can_user_access_resources" function

### PR DESCRIPTION
**Pull request by paulineribeyre**
Link to JIRA ticket if there is one: https://ctds-planx.atlassian.net/browse/MIDRC-847

Reusing the [Requestor logic](https://github.com/uc-cdis/requestor/blob/41dbe7f/src/requestor/arborist.py#L43) and making it more generic

### New Features
- Add `can_user_access_resources` function which uses a user's auth mapping to determine which of the provided resource paths they have access to

### Improvements
- The `auth_mapping` function now supports providing a username, an access token or neither (anonymous request) to match what the Arborist endpoint supports
- Use GitHub workflows to run unit tests and publish new versions to Pypi, instead of travis-ci which stopped working
- Change the repository structure from `/python/src/gen3authz/<python files>` to `/src/gen3authz/<python files>` to match other Gen3 repositories and because the shared GitHub workflows do not support the old structure

### Dependency updates
- Update dependencies

### CODE REVIEW

File: src/gen3authz/client/arborist/base.py [hunk 1]
```diff
@@ -282,16 +283,32 @@ async def healthy(self, timeout=1):
...
+        Args:
+            jwt (str): user's valid jwt access token
+            username (str): username
+
         Return:
             dict: response JSON from arborist
         """
-        data = {"username": username}
-        response = await self.post(self._auth_url.rstrip("/") + "/mapping", json=data)
+        assert not (username and jwt), "'username' and 'jwt' cannot both be provided"
```
Comment (nss10) [inline src/gen3authz/client/arborist/base.py:298]: The language seems a little confusing to me. Can we add something like "'Username' and 'jwt' must not be provided together."?

File: src/gen3authz/client/arborist/base.py [hunk 2]
```diff
@@ -282,16 +283,32 @@ async def healthy(self, timeout=1):
...
+        if username:
+            data = {"username": username}
+            response = await self.post(
+                self._auth_url.rstrip("/") + "/mapping",
+                json=data,
```
Comment (nss10) [inline src/gen3authz/client/arborist/base.py:303]: If we're using `data` variable only once, can't we use it directly in the function call?

File: src/gen3authz/client/arborist/base.py [hunk 3]
```diff
@@ -282,16 +283,32 @@ async def healthy(self, timeout=1):
...
+            )
+        elif jwt:
+            response = await self.post(
+                self._auth_url.rstrip("/") + "/mapping",
+                headers={"Authorization": f"bearer {jwt}"},
+            )
+        else:
+            response = await self.post(self._auth_url.rstrip("/") + "/mapping")
```
Comment (nss10) [inline src/gen3authz/client/arborist/base.py:311]: What is expected to happen if neither of `username` or `jwt` is passed? Do we just make the POST call without any authN?

File: tests/arborist/test_urls.py [hunk 4]
```diff
@@ -205,3 +206,68 @@ async def test_update_user_raises_error(
...
+            resources={"/a": {"service": "service1", "method": "read"}},
+        )
+
+    mock_arborist_request(
+        {
+            "/auth/mapping": {
+                "POST": (
+                    200,
+                    {
+                        "/a": [{"service": "service1", "method": "read"}],
```
Comment (nss10) [inline tests/arborist/test_urls.py:242]: Can we have any test cases involving the `*` scenarios for service or method?

**Review by nss10 — COMMENTED** [review]
Great work! Got some minor feedback. Can approve once answered/fixed.

File: src/gen3authz/client/arborist/base.py (hunk 3 above)
Comment (paulineribeyre) [reply src/gen3authz/client/arborist/base.py:311]: it's an anonymous call and arborist will only return anonymous access

File: tests/arborist/test_urls.py (hunk 4 above)
Comment (paulineribeyre) [reply tests/arborist/test_urls.py:240]: updated

File: src/gen3authz/client/arborist/base.py (hunk 2 above)
Comment (paulineribeyre) [reply src/gen3authz/client/arborist/base.py]: updated

File: src/gen3authz/client/arborist/base.py (hunk 1 above)
Comment (paulineribeyre) [reply src/gen3authz/client/arborist/base.py]: updated

**Pull request merged**